src/test/java/test/BaseTest.java

```diff
@@ -0,0 +1,42 @@
+package test;
+
+import org.openqa.selenium.*;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.testng.annotations.*;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public abstract class BaseTest {
+
+    public Logger logger = Logger.getGlobal();
+
+    public  WebDriver driver;
+
+   @BeforeTest
+    @Parameters("browser")
+    public void setup(String browser) throws Exception{
+        if(browser.equalsIgnoreCase("Firefox")){
+            System.setProperty("webdriver.gecko.driver", "src/test/resources/geckodriver.exe");
+            driver = new FirefoxDriver();
+        }
+        else if(browser.equalsIgnoreCase("Chrome")){
+            System.setProperty("webdriver.chrome.driver","src/test/resources/chromedriver.exe");
+            driver = new ChromeDriver();
+        }
+        else if(browser.equalsIgnoreCase("Edge")){
+            System.setProperty("webdriver.edge.driver","src/test/resources/msedgedriver.exe");
+            driver = new EdgeDriver();
+        }
+        else{
+            throw new Exception("Browser is not correct");
+        }
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
```

Comment:
For what you need this 10 seconds implicit wait?

Reply:
Just left how it was in an example, not sure that it is a proper time and if I need it at all...
Reduced to 5 )